### PR TITLE
SB-74: Add Reservation on FiresStore by Calendar

### PR DESCRIPTION
**Pull request by EzeLamar**
Changes:
* Allows adding reservations to Firestore by the Reservation Calendar.
* Create NewReservationType to represent the draft Reservation without the id.

### CODE REVIEW

**Comment by EzeLamar** [comment]
[Add reservations on firebase with Reservations Calendar](https://trello.com/c/3b6xf5Ct/74-add-reservations-on-firebase-with-reservations-calendar)

File: app/components/UI/Calendar/Calendar.tsx [hunk 1]
```diff
@@ -125,24 +127,33 @@ export default function Calendar({
...
+					};
+					setEvents([...myEvents, event]);
+					setShowModal(false);
+					resolve(true);
+				});
+			} else {
+				// TODO: Service call to Edit the Reservation on FireStore
+				setShowModal(false);
+				resolve(true);
+			}
```
Comment (EzeLamar) [inline app/components/UI/Calendar/Calendar.tsx:151]: This block was reserved to use the edit reservation service in a future PR.

File: app/components/Reservations/Reservation.tsx [hunk 2]
```diff
@@ -3,6 +3,14 @@ import Link from 'next/link';
...
 
 export type ReservationType = {
+	id: string;
+	court: DocumentReference;
+	owner: string;
+	startTime: Date;
+	endTime: Date;
+};
+
+export type NewReservationType = {
```
Comment (lubualo) [inline app/components/Reservations/Reservation.tsx:13]: I suggest that a reservation that has not been created yet could be called `reservationDraft`. Reservation draft won't have an ID as it hasn't been persisted in the DB yet, opposed to the normal reservation which exists in the DB

**Review by lubualo — CHANGES_REQUESTED** [review]

File: app/components/Reservations/Reservation.tsx (hunk 2 above)
Comment (EzeLamar) [reply app/components/Reservations/Reservation.tsx]: Good catch! Fixed!

Comment (lubualo) [reply app/components/Reservations/Reservation.tsx]: Awesome

**Review by lubualo — APPROVED** [review]
Good job!

**Pull request merged**